packages/contracts/hooks/ServiceNodeRewards.tsx

```diff
@@ -294,3 +298,83 @@ export function useGetRecipients({ address }: { address: Address }) {
     ...rest,
   };
 }
+
+export function useClaimThreshold({ enabled }: { enabled?: boolean }) {
+  const { data: claimThreshold, ...rest } = useContractReadQuery({
+    contract: 'ServiceNodeRewards',
+    functionName: 'claimThreshold',
+    enabled,
+  });
+
+  return {
+    claimThreshold,
+    ...rest,
+  };
+}
+
+export function useClaimCycle({ enabled }: { enabled?: boolean }) {
+  const { data: claimCycle, ...rest } = useContractReadQuery({
+    contract: 'ServiceNodeRewards',
+    functionName: 'claimCycle',
+    enabled,
+  });
+
+  return {
+    claimCycle,
+    ...rest,
+  };
+}
+
+export function useCurrentClaimTotal({ enabled }: { enabled?: boolean }) {
+  const { data: currentClaimTotal, ...rest } = useContractReadQuery({
+    contract: 'ServiceNodeRewards',
+    functionName: 'currentClaimTotal',
+    enabled,
+  });
+
+  return {
+    currentClaimTotal,
+    ...rest,
+  };
+}
+
+export function useCurrentClaimCycle({ enabled }: { enabled?: boolean }) {
+  const { data: currentClaimCycle, ...rest } = useContractReadQuery({
+    contract: 'ServiceNodeRewards',
+    functionName: 'currentClaimCycle',
+    enabled,
+  });
+
+  return {
+    currentClaimCycle,
+    ...rest,
+  };
+}
+
+export function useClaimCycleDetails({ enabled }: { enabled?: boolean }) {
+  const claimThresholdHook = useClaimThreshold({ enabled });
+  const claimCycleHook = useClaimCycle({ enabled });
+  const currentClaimTotalHook = useCurrentClaimTotal({ enabled });
+
+  const status = mergeContractReadStatuses(
```

Comment:
You could make mergeContractReadStatuses take in an array of status and do array operations to merge them? Saves the extra function call and is more versatile.

Reply:
that would require array operations, so this is cheaper